src/pyhf/constraints.py

```diff
@@ -39,57 +55,70 @@ def _precompute(self):
                 normal_constraint_sigmas.append([1.0] * len(thisauxdata))
 
             normal_constraint_data.append(thisauxdata)
-            normal_constraint_mean_indices.append(self.par_indices[parslice])
-
-        if normal_constraint_mean_indices:
-            normal_mean_idc = default_backend.concatenate(
-                list(
-                    map(
-                        lambda x: default_backend.astensor(x, dtype='int'),
-                        normal_constraint_mean_indices,
-                    )
-                )
-            )
-            normal_sigmas = default_backend.concatenate(
-                list(map(default_backend.astensor, normal_constraint_sigmas))
-            )
-            normal_data = default_backend.concatenate(
-                list(
-                    map(
-                        lambda x: default_backend.astensor(x, dtype='int'),
-                        normal_constraint_data,
-                    )
-                )
+
+        if self.parameter_helper.index_selection:
+            normal_sigmas = default_backend.concatenate(normal_constraint_sigmas)
+
+            normal_data = default_backend.astensor(
+                default_backend.concatenate(normal_constraint_data), dtype='int'
             )
 
-            self.normal_data = tensorlib.astensor(
+            self._normal_data = default_backend.astensor(
                 default_backend.tolist(normal_data), dtype='int'
             )
-            self.normal_sigmas = tensorlib.astensor(
+            self._normal_sigmas = default_backend.astensor(
                 default_backend.tolist(normal_sigmas)
             )
-            self.normal_mean_idc = tensorlib.astensor(
-                default_backend.tolist(normal_mean_idc), dtype='int'
-            )
         else:
-            self.normal_data, self.normal_sigmas, self.normal_mean_idc = (
-                None,
-                None,
-                None,
-            )
+            self._normal_data, self._normal_sigmas = (None, None)
```

Comment:
self.normal_mean_idc is gone?

Reply:
yes this is handled by self.access_field